### PR TITLE
chore(query): update opendal version and add retry io timeout

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

chore(query): update opendal version and add retry io timeout

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14496)

### CODE REVIEW

File: src/common/storage/src/operator.rs [hunk 1]
```diff
@@ -102,15 +102,32 @@ pub fn build_operator<B: Builder>(builder: B) -> Result<Operator> {
...
+            }
+
+            if retry_io_timeout != 0 {
+                // Return timeout error if the io operation timeout
+                timeout_layer.with_io_timeout(Duration::from_secs(retry_io_timeout));
+            }
+
+            // Return timeout error if the request speed is less than
+            // 1 KiB/s.
+            timeout_layer.with_speed(1024)
```
Comment (Xuanwo) [inline src/common/storage/src/operator.rs:129]: This is not used anymore. We can remove this.

File: Cargo.lock [hunk 2]
```diff
@@ -9109,6 +9109,36 @@ name = "opendal"
...
+ "serde_json",
+ "sha2",
+ "tokio",
+ "uuid",
+]
+
+[[package]]
+name = "opendal"
+version = "0.44.2"
+source = "git+https://github.com/apache/opendal.git?rev=e94f068#e94f068a9977fcd20c9bb3626aaaa8be833209a8"
```
Comment (Xuanwo) [inline Cargo.lock:9141]: Seems we are refering to two version of opendal. Please use `repalce` instead change `opendal` directly.

File: Cargo.toml [hunk 3]
```diff
@@ -110,7 +109,7 @@ members = [
 # databend maintains
 jsonb = { git = "https://github.com/datafuselabs/jsonb", rev = "e21b273" }
 
-opendal = { version = "0.44.2", features = [
+opendal = { git = "https://github.com/apache/opendal.git", rev = "e94f068", features = [
```
Comment (Xuanwo) [inline Cargo.toml:112]: This line not needed.

File: Cargo.toml
Comment (Xuanwo) [inline Cargo.toml:1]: Add this line under `[patch.crates-io]`

```rust
opendal = { git = "https://github.com/apache/opendal", rev = "commit" }
```

**Review by Xuanwo — APPROVED** [review]
Thanks!

**Review by dantengsky — APPROVED** [review]

**Pull request merged**